src/node.cc

```diff
@@ -3615,6 +3611,365 @@ static void PrintHelp() {
 }
 
 
+static void CheckIfAllowedInNodeopt(const char* exe, bool is_nodeopt,
+                                    const char* arg) {
+  if (!is_nodeopt)
+    return;
+
+  // Find the arg prefix when its --some_arg=val
+  const char* eq = strchr(arg, '=');
+  size_t arglen = eq ? eq - arg : strlen(arg);
+
+  static const char* whitelist[] = {
+    // Node options
+    "-r", "--require",
+    "--no-deprecation",
+    "--no-warnings",
+    "--trace-warnings",
+    "--redirect-warnings",
+    "--trace-deprecation",
+    "--trace-sync-io",
+    "--trace-events-enabled",
+    "--track-heap-objects"
+    "--throw-deprecation",
+    "--prof-process",
+    "--zero-fill-buffers",
+    "--v8-pool-size",
+    "--use-openssl-ca",
+    "--use-bundled-ca",
+    "--enable-fips",
+    "--force-fips",
+    "--openssl-config",
+    "--icu-data-dir",
+
+    // V8 options
+    "--page_promotion",
```

Comment:
These are basically the options that don't enable any extra APIs (code that needs extra APIs will have already arranged for them to be enabled), and that can be used to tweak runtime performance and/or observe the V8 runtime behaviour to diagnose why live code has particular performance problems. Feel free to comment on any that should be added/removed from whitelist.